src/libraries/System.Collections/tests/Generic/SortedDictionary/SortedDictionary.Generic.Tests.cs

```diff
@@ -62,6 +62,18 @@ public void SortedDictionary_Generic_Constructor_IDictionary_IComparer(int count
             Assert.Equal(comparer, copied.Comparer);
         }
 
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void SortedDictionary_Generic_Constructor_SortedDictionary_IComparer(int count)
+        {
+            IComparer<TKey> comparer = GetKeyIComparer();
+            IDictionary<TKey, TValue> source = GenericIDictionaryFactory(count);
+            SortedDictionary<TKey, TValue> sourceSorted = new SortedDictionary<TKey, TValue>(source, comparer);
+            SortedDictionary<TKey, TValue> copied = new SortedDictionary<TKey, TValue>(sourceSorted, comparer);
+            Assert.Equal(sourceSorted, copied);
+            Assert.Equal(comparer, copied.Comparer);
+        }
+
```

Comment:
Do we have tests for when the collection passed in does not have the same comparer as the one passed in? For both IDictionary<K,V> and SortedDictionary<K,V> ?  I don't see them.

Reply:
Dictionary classes only have constructors with IEqualityComparer<TKey> not IComparer<TKey>. In SortedDictionary are we okay using a inline comparer which return negative of what GetKeyIComparer returns?